framework/post_processors/lua/lua_PPPrinterSetOptions.cc

```diff
@@ -0,0 +1,177 @@
+                              "vertical",
+                              "The table format with which to print scalar "
+                              "PostProcessors");
+
+  using namespace chi_data_types;
+  params.ConstrainParameterRange(
+    "scalar_pp_table_format",
+    AllowableRangeList::New({"horizontal", "vertical"}));
+
+  params.AddOptionalParameterArray(
+    "events_on_which_to_print_postprocs",
+    std::vector<std::string>{"SolverInitialized",
+                             "SolverAdvanced",
+                             "SolverExecuted",
+                             "ProgramExecuted"},
+    "A list of events on which to print post-processors");
+
+  params.AddOptionalParameter(
+    "print_scalar_time_history",
+    true,
+    "Controls whether a time history of scalar post-processors are printed. If "
+    "false, only the latest version will be printed.");
+
+  params.AddOptionalParameter(
+    "print_vector_time_history",
+    true,
+    "Controls whether a time history of vector post-processors are printed. If "
+    "false, only the latest version will be printed.");
+
+  params.AddOptionalParameter("per_column_size_scalars",
+                              true,
+                              "Controls the sizing of printed columns. If "
+                              "false all the columns will be the same size.");
+
+  params.AddOptionalParameter("per_column_size_vectors",
+                              true,
+                              "Controls the sizing of printed columns. If "
+                              "false all the columns will be the same size.");
+
+  params.AddOptionalParameter(
+    "table_column_limit",
+    120,
+    "The maximum column, if reached, would cause tables to be wrapped. A "
+    "minimum limit of 80 is automatically enforced.");
+
+  params.AddOptionalParameter(
+    "time_history_limit",
+    15,
+    "Maximum amount of time values to show in post-processor histories. A "
+    "maximum of 1000 is automatically enforced.");
+
+  params.AddOptionalParameter(
+    "csv_filename",
+    "",
+    "If not empty, a file will be printed with all the post-processors "
+    "formatted as comma seperated values.");
+
+  return params;
+}
+
+InputParameters GetSyntax_PPPrinterSetOptions()
+{
+  InputParameters params;
+
+  params.SetGeneralDescription(
+    "Wrapper function to set options of the singleton PostProcessorPrinter.");
+  params.SetDocGroup("doc_PPUtils");
+
+  params.AddRequiredParameterBlock("arg0", "Options parameter block");
+  params.LinkParameterToBlock("arg0", "chi::PostProcessorPrinterOptions");
+
+  return params;
+}
+
+ParameterBlock PostProcessorPrinterSetOptions(const InputParameters& params)
+{
+  auto& printer = PostProcessorPrinter::GetInstance();
+
+  const auto& set_params = params.ParametersAtAssignment().GetParam("arg0");
+
+  for (const auto& param : set_params)
+  {
+    const std::string param_name = param.Name();
+    if (param_name == "scalar_pp_table_format")
```

Comment:
Just a note that a `switch` on a string hash works well in cases where you end up with `if/else if` spaghetti on strings.

Reply:
That is a really good comment. Let me try it out.

Reply:
Don't know if this would be great for everyone. But put it in there.